xdsl/printer.py

```diff
@@ -729,25 +730,38 @@ def print_op_with_default_format(self, op: Operation) -> None:
         self.print(" : ")
         self.print_operation_type(op)
 
-    def print_operation_type(self, op: Operation) -> None:
+    def print_function_type(
+        self, input_types: Iterable[Attribute], output_types: Iterable[Attribute]
+    ):
         self.print("(")
-        self.print_list(op.operands, lambda operand: self.print_attribute(operand.type))
+        self.print_list(input_types, self.print_attribute)
         self.print(") -> ")
-        if len(op.results) == 0:
+
+        oi = iter(output_types)
+        try:
+            first_type = next(oi)
+        except StopIteration:
             self.print("()")
-        elif len(op.results) == 1:
-            res_type = op.results[0].type
+            return
+
+        try:
+            second_type = next(oi)
+        except StopIteration:
             # Handle ambiguous case
-            if isinstance(res_type, FunctionType):
-                self.print("(", res_type, ")")
+            if isinstance(first_type, FunctionType):
+                self.print("(", first_type, ")")
             else:
-                self.print(res_type)
-        else:
-            self.print("(")
-            self.print_list(
-                op.results, lambda result: self.print_attribute(result.type)
-            )
-            self.print(")")
+                self.print(first_type)
```

Comment:
I'm not a huge huge fan of this style, especially since it's not in a hot loop or anything.
Do you think we could either make the requirement a `Sequence`, or cast this to a list to then use a match/case that is much more readable?

Reply:
It feels like the most idiomatic way of doing things to me, and it doesn't feel particularly weird. It's a clear reflection of the logic to me, so it's not optimisation for optimisation's sake, but as a reflection of the behaviour, which seems like a good idea in general. I'd rather keep it as-is, unless you really have a strong preference to refactor it. I'm happy to add comments to make it clearer if you'd like.

Reply:
Okay, I'm fine with it if you feel more strong towards it!